vm.go

```diff
@@ -1451,14 +1451,15 @@ var inc _inc
 func (_inc) exec(vm *vm) {
 	v := vm.stack[vm.sp-1]
 
-	if i, ok := assertInt64(v); ok {
-		v = intToValue(i + 1)
-		goto end
+	switch i := v.(type) {
+	case valueInt:
+		v = i + 1
+	case valueInt64:
+		v = i + 1
+	case valueFloat:
+		v = i + 1
```

Comment:
[opt] If we want to reduce the risk/changes here I can remove this case and replace it with a default case where I would put the existing valueFloat logic.

something like this:
```
func (_inc) exec(vm *vm) {
	v := vm.stack[vm.sp-1]

	switch i := v.(type) {
	case valueInt:
		v = i + 1
	case valueInt64:
		v = i + 1
	default:
		v = valueFloat(v.ToFloat() + 1)
	}

	vm.stack[vm.sp-1] = v
	vm.pc++
}
```

I'm leaning more towards this approach but want to get your opinions.